lib/ansible/plugins/process/__init__.py

```diff
@@ -0,0 +1,98 @@
+# (c) 2017, Red Hat, Inc. <support@ansible.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import multiprocessing
+import signal
+import threading
+
+from abc import ABCMeta, abstractmethod
+from collections import deque
+
+from ansible.executor.shared_plugin_loader import SharedPluginLoaderObj
+from ansible.module_utils.six import with_metaclass
+
+__all__ = ["ProcessModelBase", "keyboard_interrupt_event", "ResultsSentinel"]
+
+
+class ResultsSentinel:
+    pass
+
+
+keyboard_interrupt_event = multiprocessing.Event()
+
+
+def keyboard_interrupt_handler(signum, frame):
+    keyboard_interrupt_event.set()
+
+
+class ProcessModelBase(with_metaclass(ABCMeta, object)):
```

Comment:
you might want to inherit from AnsiblePlugin, among other things, this will give you 'option processing' for free, not needed in current plugins but might want for future